### PR TITLE
Including Bulk Authentication Delete.

### PR DESCRIPTION
**Pull request by pranavrao1**
This commit is to include a new request and response message for a bulk
authorization delete for a single resource. It includes a descriptions
and the configurations to send this request and the expected succesful
and failied responses.

A typo was also in the 'Securing an XMPP Multi-User chat' flow.

### CODE REVIEW

**Review by balthorium — COMMENTED** [review]
Nm, see comment below.

File: draft-abiggs-saag-key-management-service.md [hunk 1]
```diff
@@ -1599,7 +1603,7 @@ JWE(K_ephemeral, {
 
 ### Delete Authorization
 
-To remove an authorization from a KMS resource object, any user currently authorized on the same resource object may issue a delete authorization request.  The request message conforms to the basic request message structure, where the method is "delete", and the URI is either that of the authorization object to be deleted, or the URI of the collection of authorizations within a particular KMS resource object appended with an authId query parameter whose value matches that of the authorization object to be deleted.
+To remove an authorization from a KMS resource object, any user currently authorized on the same resource object may issue a delete authorization request.  The request message conforms to the basic request message structure, where the method is "delete", and the URI is either that of the authorization object to be deleted, or a collection of authorizations within a particular KMS resource object appended to an authIds variable parameter whose value matches that of the authorization object to be deleted.
```
Comment (balthorium) [inline draft-abiggs-saag-key-management-service.md:1607]: I don't quite understand this phrase: "a collection of authorizations within a particular KMS resource object appended to an authIds variable parameter whose value matches that of the authorization object to be deleted."

Maybe change to something like this?

"a collection of authorizations within a particular KMS resource object accompanied by an authIds array attribute containing values that match the authorization objects to be deleted."

File: draft-abiggs-saag-key-management-service.md [hunk 2]
```diff
@@ -1644,6 +1648,27 @@ JWE(K_ephemeral, {
...
+  "method": "delete",
+  "uri": "/authorizations",
+  "authIds": {
+    "557ac05d-5751-43b4-a04b-e7eb1499ee0a",
+    "c1047d54-c485-4aa1-9704-fee07adae46c",
+    ...
+  }
+})
+~~~
+
```
Comment (balthorium) [inline draft-abiggs-saag-key-management-service.md:1671]: Looks good.

File: draft-abiggs-saag-key-management-service.md [hunk 3]
```diff
@@ -1670,6 +1695,25 @@ JWE(K_ephemeral, {
...
+    "authIds": {
+      "557ac05d-5751-43b4-a04b-e7eb1499ee0a",
+      "c1047d54-c485-4aa1-9704-fee07adae46c",
+      ...
+    }
+    "resourceUri": "/resources/7f35c3eb-95d6-4558-a7fc-1942e5f03094"
+  }
+})
+~~~
+
```
Comment (balthorium) [inline draft-abiggs-saag-key-management-service.md:1716]: Are we restricting bulk deletes to collections of authorizations on the same KRO?  If not, then the "resourceUri" attribute might not make sense here.

Comment (pranavrao1) [reply draft-abiggs-saag-key-management-service.md:1716]: We list the `uri` as a way to identify it as authorizations to delete. At least that is how it is in our implementation. Do we not need this detail here?

File: draft-abiggs-saag-key-management-service.md (hunk 1 above)
Comment (pranavrao1) [reply draft-abiggs-saag-key-management-service.md:1607]: how about this:
```
a collection of authorizations, included as an attribute authIds to a particular KMS resource object, whose value matches that of the authorization object to be deleted.
```

How is this?

Comment (balthorium) [reply draft-abiggs-saag-key-management-service.md:1607]: Hm, still not clear.  Might be necessary to break into more than one sentence.

**Pull request merged**